### PR TITLE
:sparkles: add postgresql compatibility for datastream-to-bigquery template

### LINKED ISSUE #583: [Feature Request]: Add postgresql metatada replication support for Datastream-to-BigQuery-template
### Related Template(s)

datastream-to-bigquery

### What feature(s) are you requesting?

Hi, 

I would like to implement this [tutorial](https://cloud.google.com/datastream/docs/implementing-datastream-dataflow-analytics?hl=fr) with a postgresql as source database.

As far I can see, only MySQL or Oracle metadata (which is the default) are supported, in particular in this  method : https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/99a36f26746b554ec13d049cec58be6b3af08b1d/v2/common/src/main/java/com/google/cloud/teleport/v2/values/DatastreamRow.java#L166

I will make a PR to add a case for PostgreSQL with a `_metadata_lsn` instead of `_metadata_scn`

### PR DESCRIPTION
**Pull request by rguillome**
Add PostgreSQL compatibility for the template Datastream to Bigquery :

- Add the metadata_lsn in the metadata columns
- Create a complete PostgreSQL mapping with BigQuery for the merge

Fix #583

### CODE REVIEW

File: v2/common/src/main/java/com/google/cloud/teleport/v2/utils/DataStreamClient.java [hunk 1]
```diff
@@ -120,6 +124,10 @@ public Map<String, StandardSQLTypeName> getObjectSchema(
       return getMysqlObjectSchema(streamName, schemaName, tableName, sourceConnProfile);
     } else if (sourceConnProfile.getOracleSourceConfig() != null) {
       return getOracleObjectSchema(streamName, schemaName, tableName, sourceConnProfile);
+    } else if (sourceConnProfile.getOracleSourceConfig() != null) {
+      return getOracleObjectSchema(streamName, schemaName, tableName, sourceConnProfile);
```
Comment (bvolpato) [inline v2/common/src/main/java/com/google/cloud/teleport/v2/utils/DataStreamClient.java:128]: minor: duplicated if

**Comment by bvolpato** [comment]
Thanks for contributing @rguillome. 

@nirfi Can you please take a look / delegate?

File: v2/common/src/main/java/com/google/cloud/teleport/v2/utils/DataStreamClient.java (hunk 1 above)
Comment (rguillome) [reply v2/common/src/main/java/com/google/cloud/teleport/v2/utils/DataStreamClient.java]: Sorry, removed in the 2nd commit

**Comment by nirfir** [comment]
Hey @rguillome , thanks for the contriubtion! We'll review this in the upcoming few days.

Nir

File: v2/common/src/main/java/com/google/cloud/teleport/v2/utils/DataStreamClient.java [hunk 2]
```diff
@@ -458,4 +533,57 @@ public StandardSQLTypeName convertMysqlToBigQueryColumnType(MysqlColumn column)
...
+      case "CHARACTER":
+      case "VARCHAR":
+      case "TEXT":
+      case "TINYTEXT":
+      case "MEDIUMTEXT":
+      case "LONGTEXT":
+        return StandardSQLTypeName.STRING;
+      case "BOOLEAN":
+        return StandardSQLTypeName.BOOL;
+      case "TINYINT":
```
Comment (nirfir) [inline v2/common/src/main/java/com/google/cloud/teleport/v2/utils/DataStreamClient.java:552]: "OID" is another type that should be mapped to INT64

**Comment by rguillome** [comment]
Hi @bvolpato , 
is there any chance this would be merged soon ?
Cheers,

**Comment by bvolpato** [comment]
Circling back to @nirfir. Changes looks good to me but Nir has been looking at those templates much closer than me.

**Comment by giriraj-hh** [comment]
@bvolpato , @nirfir -- Hello, could you please let me know if this would be available sooner? We are actively looking to integrate this dataflow template for CDC(via Datastream) plus data transformations from the source Cloud Sql for postgresql to Bigquery.

Regards,
Giriraj

**Comment by SargisPlusPlus** [comment]
Hey @rguillome 
Thank you for making this change.
I am still unable to use this template with postgres.

ie: https://github.com/GoogleCloudPlatform/DataflowTemplates/issues/2025

For some reason, Oracle columns `_metadata_scn` are created instead of `_metadata_lsn`.

**Comment by manojredward** [comment]
Hello @rguillome @bvolpato @nirfir 

Is this issue resolved? I am also facing an issue while implementing Datastream To BigQuery with JavaScript UDF Functions.

Here is the error details,

Merge Job Failed With BigQuery Exception: com.google.cloud.bigquery.BigQueryException: Query error: Unrecognized name: _metadata_lsn; Did you mean _metadata_ssn? at [1:937] Statement: BEGIN BEGIN TRANSACTION; MERGE `<table_name>`

**Pull request merged**